core/bindings.rs

```diff
@@ -929,10 +929,21 @@ fn encode(
       return;
     }
   };
-  let text_str = text.to_rust_string_lossy(scope);
-  let zbuf: ZeroCopyBuf = text_str.into_bytes().into();
 
-  rv.set(to_v8(scope, zbuf).unwrap())
+  let capacity = text.utf8_length(scope);
```

Comment:
I kind of already know this to be a not-good-idea but:

`v8::String` can (if externalizable) be turned into `v8::String::ExternalStringresource` which can be used to to get to get the actual string buffer data. Now that buffer can be used to create a `v8::BackingStore` and then sent onwards to a `v8::ArrayBuffer`. => Zero copy encoding.

Of course, this means that the AB now owns the string buffer and I'm very much unsure if the original `v8::String` can deal with that, much less what happens if the AB now makes changes to the buffer data.

Reply:
@aapoalas This doesn't work, because the string is internally stored as WTF-16 by V8, not UTF-8. Conversion always needs to happen.

Reply:
Ah, very good point. It was a bad idea anyway due to the various potential corner cases mentioned :)